File: src/test/resources/protocols/test/java/io/vlingo/gradle/actortest/Test2ProtocolActor.java
package io.vlingo.gradle.actortest;

import java.util.Arrays;

import io.vlingo.actors.Actor;

public class Test2ProtocolActor extends Actor implements Test2Protocol {

    private static String doOneThing;
    public static String getDoOneThing() { return doOneThing; }
    public static void setDoOneThing(String value) { doOneThing = value; }

    private static String doAnotherThingUsing;
    public static String getDoAnotherThingUsing() { return doAnotherThingUsing; }
    public static void setDoAnotherThingUsing(String value) { doAnotherThingUsing = value; }

    private static String somethingRatherWonderful;
    public static String getSomethingRatherWonderful() { return somethingRatherWonderful; }
    public static void setSomethingRatherWonderful(String value) { somethingRatherWonderful = value; }

    @Override
    public void doOneThing() {
        doOneThing = "doOneThing";
    }

    @Override
    public void doAnotherThingUsing(final String text, final int value) {
        doAnotherThingUsing = "doAnotherThingUsing: " + text + value;
    }

    @Override
    public void somethingRatherWonderful(final Test1Protocol test1) {
        test1.doSomethingElseWith(1);
        test1.doSomethingWith("testing", Arrays.asList("one", "two", "three"));
        somethingRatherWonderful = "somethingRatherWonderful: " + Test1ProtocolActor.getDoSomethingElseWithValue() + Test1ProtocolActor.getDoSomethingWith();
    }
}

File: src/test/resources/protocols/test/kotlin/io/vlingo/gradle/actortest/Test2ProtocolActor.kt
@file:JvmName("Test2ProtocolActor")
package io.vlingo.gradle.actortest

import io.vlingo.actors.Actor

class Test2ProtocolActor : Actor(), Test2Protocol {

    companion object {

        private var doOneThing: String? = null
        @JvmStatic fun getDoOneThing() = doOneThing
        @JvmStatic fun setDoOneThing(value: String?) { doOneThing = value }

        private var doAnotherThingUsing: String? = null
        @JvmStatic fun getDoAnotherThingUsing() = doAnotherThingUsing
        @JvmStatic fun setDoAnotherThingUsing(value: String?) { doAnotherThingUsing = value }

        private var somethingRatherWonderful: String? = null
        @JvmStatic fun getSomethingRatherWonderful() = somethingRatherWonderful
        @JvmStatic fun setSomethingRatherWonderful(value: String?) { somethingRatherWonderful = value }
    }

    override fun doOneThing() {
        doOneThing = "doOneThing";
    }

    override fun doAnotherThingUsing(text: String, value: Int) {
        doAnotherThingUsing = "doAnotherThingUsing: $text$value";
    }

    override fun somethingRatherWonderful(test1: Test1Protocol) {
        test1.doSomethingElseWith(1);
        test1.doSomethingWith("testing", listOf("one", "two", "three"));
        somethingRatherWonderful = "somethingRatherWonderful: " + Test1ProtocolActor.getDoSomethingElseWithValue() + Test1ProtocolActor.getDoSomethingWith()
    }
}
